copy/multiple.go

```diff
@@ -87,17 +89,43 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 	}
 
 	// Copy each image, or just the ones we want to copy, in turn.
+	optionsInstances := options.Instances
 	instanceDigests := updatedList.Instances()
+	if len(options.InstancePlatforms) > 0 {
+		// Build a map for helping us avoid duplication of instance digests.
+		instanceDigestMap := make(map[digest.Digest]struct{})
+		for _, instanceDigest := range optionsInstances {
+			instanceDigestMap[instanceDigest] = struct{}{}
+		}
+		// Choose the best match for each platform we were asked to also copy.
+		for _, platform := range options.InstancePlatforms {
+			platformContext := types.SystemContext{
+				OSChoice:           platform.OS,
+				ArchitectureChoice: platform.Architecture,
+				VariantChoice:      platform.Variant,
+			}
+			instanceDigest, err := updatedList.ChooseInstance(&platformContext)
+			if err != nil {
+				return nil, fmt.Errorf("While choosing the instance for platform spec %q: %w", platform, err)
+			}
+			instanceDigestMap[instanceDigest] = struct{}{}
+		}
+		// Deduplicate the set of digests.
+		optionsInstances = nil
```

Comment:
It’s might be a bit misleading to name this `optionsInstances` when it depends on the other option as well.

Arguably `optionsInstances` includes `options.Instances*`, but it seems simple to avoid this uncertainty.

Maybe `[instance]digestsToCopy`? That would also say what the data is for.

Reply:
The set is tied to the `CopySpecificImages` choice, so how about `specificImages`?